### ISSUE TITLE
Select PHP71 does not install PHP 7.1

### ISSUE THREAD

**Issue by mikeerickson** [laradock/laradock#776]
If you select the `PHP71` option in the `.ENV` file, it still installed `PHP 7.0.x`

**Comment by mikeerickson** [comment]
As outlined in `.ENV`

```
### PHP version (Does not apply for HHVM)
# PHP_VERSION=56
# PHP_VERSION=70
PHP_VERSION=71
```

**Comment by mikeerickson** [comment]
@Mahmoudz If you have any free time, could you please review this issue.  It all looks right to me, but I am not sure if there is something else causing the problem.

**Comment by bestlong** [comment]
if use PHP56, in workspace container also install 7.0

**Comment by winfried-van-loon** [comment]
To fix this, I've opened a pull request in the Workspace container: laradock/workspace/pull/15 However, I'm not sure how to test my changes.

**Comment by winfried-van-loon** [comment]
This issue is waiting on laradock/workspace#15

**Comment by Dayjo** [comment]
Hello, just adding my interest to the issue, also need to be able to use 7.1. 

Happy to help test if I can.

**Comment by winfried-van-loon** [comment]
Sadly I'm a bit stuck as I don't know how to test the base image and how to get it on Docker Hub. Details are here: laradock/workspace#15

**Comment by bestlong** [comment]
Hi @winfried-van-loon, you can build image by you self
at laradock/workspace project root run
```
docker build -t laradock/workspace:test-5.6 -f Dockerfile-56 .
```
if success, go to laradock/laradoc project modify workspace/Dockerfile-56 base image name
```
FROM laradock/workspace:1.3
```
change to
```
FROM laradock/workspace:test-5.6
```
then run (.env set PHP_VERSION=56)
```
docker-compose build workspace
```
after build success
```
docker-compose up -d nginx mysql
docker-compose exec workspace php -v
```
you will get PHP 5.6 version message.

**Comment by winfried-van-loon** [comment]
Interesting! I'll try it soon and report back here. Thanks!

**Comment by mikeerickson** [comment]
@bestlong So, assuming I want to build a workspace container with PHP 7, I can replace reference of 56 to use 70?

**Comment by bestlong** [comment]
@mikeerickson Yes

**Comment by dtunes** [comment]
Hi guys,

the workspace base image always installs php70, if you check the repo https://github.com/laradock/workspace, you'll notice the Dockerfile-71 I'm not sure if its finished, but its not setup in the same way as the php-fpm containers, if you look into the workspace directory, the Dockerfile-<PHP_VERSION> all use the same base image workspace:1.3, php-fpm containers pull in specific versions e.g.

laradock/php-fpm:5.6--1.4
laradock/php-fpm:7.0--1.4
laradock/php-fpm:7.1--1.4

where's all workspace Dockerfiles-<PHP_VERSION> are based on:
laradock/workspace:1.3

however, the php-fpm/Dockerfile-71 doesn't look finished, but I haven't checked everything.

with PHP_VERSION=56
![image](https://cloud.githubusercontent.com/assets/1164234/25210009/28bacdfa-25d2-11e7-9730-ba4a568108d3.png)

with PHP_VERSION=70
![image](https://cloud.githubusercontent.com/assets/1164234/25210110/af0bce54-25d2-11e7-80eb-63e5d3834d68.png)

with PHP_VERSION=71
![image](https://cloud.githubusercontent.com/assets/1164234/25210404/b73916da-25d3-11e7-8be0-45603700c135.png)

As you can see, it's quite confusing, I haven't had the time to explore it further.

This is not a fix obviously, maybe even highlighting the obvious to some people, but we need to fix this. I'm currently working on a new container as you may notice in the branch name blackfire.io. I'm not gonna have the time to work on this until probably May, hopefully someone else will find the time to do it soon, otherwise, I'll have another look later "probably May".

**Comment by FairHypo** [comment]
Meet same problem with laradock.
I changed all possible options, removed containers, rebuilt it again etc...
But nothing is working.
Can you help me to fix that issue?

**Comment by dtunes** [comment]
Just use PHP_VERSION=70 for now.

**Comment by FairHypo** [comment]
"Just use PHP_VERSION=70 for now."

But i need php 7.1 for my project.

**Comment by FairHypo** [comment]
upd.

After last git pull a new issue become:

`Traceback (most recent call last):
  File "bin/docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 64, in main
  File "compose/cli/main.py", line 116, in perform_command
  File "compose/cli/main.py", line 876, in up
  File "compose/project.py", line 392, in up
  File "compose/project.py", line 182, in get_services_without_duplicate
  File "compose/project.py", line 170, in get_services
  File "compose/project.py", line 141, in get_service
  File "compose/project.py", line 608, in __init__
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
Failed to execute script docker-compose
`

**Comment by mikeerickson** [comment]
@FairHypo If you use PHP_VERSION=70 it will actually install PHP 7.1

**Comment by Mahmoudz** [comment]
@mikeerickson @winfried-van-loon any updates on this one?! is it still occurring? and shall we make 7.1 our default version!?

**Comment by winfried-van-loon** [comment]
Yep! Had to finish https://github.com/laradock/workspace/issues/8 to be able to get to these points. Now that that's done, today or tomorrow, this issue and the thee similair issues will be fixed.

**Comment by FairHypo** [comment]
" If you use PHP_VERSION=70 it will actually install PHP 7.1"

https://i.gyazo.com/8ba404054d2b2bd000b861e7d767a981.png

r you kiddin me? =)

Hope that today or tomorrow fixes will be helpful. Thx all for answers)

**Comment by Mahmoudz** [comment]
@FairHypo yes it sucks! but be patient, most of us are busy people in general.
If you need this so bad and I can totally understand, try to fix it and submit a PR when done we'll merge it quickly. Issues like this are usually very simple to fix.
Regards,

**Comment by Mahmoudz** [comment]
Solved. _(requires upgrade and re-build)._

**Comment by marknt15** [comment]
hi @Mahmoudz newbie here, how can I upgrade my laradock version? Our laradock is outdated and I can't use **PHP 7.1**. I tried changing it in the .env file and rebuilt the php-fpm container with **--no-cache** but it is still version 5.6 when I go inside my container and do a **php -v**. Thank you

**Comment by DiegoFelipe** [comment]
after change the version in php-fpm, also build workspace then it will work
